src/metabase-report-executor/src/main/java/com/champtitles/metabasereportexecutor/executor/MetabaseClient.java

```diff
@@ -0,0 +1,245 @@
+                .header("Content-Type", "application/json")
+                .POST(createBody(sessionRequest))
+                .build();
+        String response = sendHttpRequest(httpRequest, 200);
+
+        try {
+            SessionResponse sessionResponse = objectMapper.readValue(response, SessionResponse.class);
+            logger.info("logged in successfully");
+            logger.debug("session id: {}", sessionResponse.id());
+            sessionId = sessionResponse.id();
+            return sessionResponse.id();
+
+        } catch (JsonProcessingException e) {
+            logger.error("failed to parse session properties response: {}", response);
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * Create a Card in Metabase
+     * <p>
+     * metabase.com/glossary/card
+     * metabase.com/docs/latest/api/card#post-apicard
+     *
+     * @param name name of the card to create
+     * @return id of the created card
+     */
+    public String createCard(String name) {
+        if (StringUtils.isBlank(sessionId)) {
+            throw new RuntimeException("you must login before creating a card");
+        }
+
+        CreateCardRequest createCardRequest = new CreateCardRequest(name);
+
+        HttpRequest httpRequest = HttpRequest
+                .newBuilder()
+                .uri(createUri("/api/card"))
+                .header("Content-Type", "application/json")
+                .header("X-Metabase-Session", sessionId)
+                .POST(createBody(createCardRequest))
+                .build();
+        String response = sendHttpRequest(httpRequest, 202);
+
+        try {
+            CreateCardResponse createCardResponse = objectMapper.readValue(response, CreateCardResponse.class);
+            logger.info("card created successfully: {}", createCardResponse.id());
+            return createCardResponse.id();
+
+        } catch (JsonProcessingException e) {
+            logger.error("failed to parse create card response: {}", response);
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * Execute a query on a Metabase Card and return the results as XLSX data
+     * <p>
+     * metabase.com/docs/latest/api/card#post-apicardcard-idqueryexport-format
+     *
+     * @param cardId metabase card to query
+     * @return XLSX data
+     */
+    public String queryCardGetXlsx(String cardId) {
+        if (StringUtils.isBlank(sessionId)) {
+            throw new RuntimeException("you must login before querying a card");
+        }
+
+        HttpRequest httpRequest = HttpRequest
+                .newBuilder()
+                .uri(createUri("/api/card/" + cardId + "/query/xlsx"))
+                .header("X-Metabase-Session", sessionId)
+                .POST(HttpRequest.BodyPublishers.noBody())
+                .build();
+        String response = sendHttpRequest(httpRequest, 200);
+        logger.info("query card response size: {} bytes", response.length());
```

Comment:
standard style would be to name the value first, basically following the object toString format. No change required.

```suggestion
        logger.info("query card response size: bytes={}", response.length());
```